src/commands/lightning/preview/app.ts

```diff
@@ -1,41 +1,137 @@
 /*
- * Copyright (c) 2023, salesforce.com, inc.
+ * Copyright (c) 2024, salesforce.com, inc.
  * All rights reserved.
  * Licensed under the BSD 3-Clause license.
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
 import { SfCommand, Flags } from '@salesforce/sf-plugins-core';
-import { Messages } from '@salesforce/core';
+import { Messages, Logger, Org } from '@salesforce/core';
+import { PreviewUtils } from '@salesforce/lwc-dev-mobile-core';
+import { OrgUtils } from '../../../shared/orgUtils.js';
 
 Messages.importMessagesDirectoryFromMetaUrl(import.meta.url);
 const messages = Messages.loadMessages('@salesforce/plugin-lightning-dev', 'lightning.preview.app');
 
-export type LightningPreviewAppResult = {
-  path: string;
+enum Platform {
+  desktop = 'desktop',
+  ios = 'ios',
+  android = 'android',
+}
+
+type LightningPreviewAppFlags = {
+  name: string | undefined;
+  'target-org': Org;
+  'device-type': Platform;
+  'device-id': string | undefined;
 };
 
-export default class LightningPreviewApp extends SfCommand<LightningPreviewAppResult> {
+export default class LightningPreviewApp extends SfCommand<void> {
   public static readonly summary = messages.getMessage('summary');
   public static readonly description = messages.getMessage('description');
   public static readonly examples = messages.getMessages('examples');
 
   public static readonly flags = {
     name: Flags.string({
       summary: messages.getMessage('flags.name.summary'),
```

Comment:
As per latest discussion [here](https://salesforce.quip.com/lw7LAqDyuhjn#TUKADArZyof), `--name` is now optional.